elastic/beats
# [Bug] fix high IO after sudden filebeat stop (#35893)

## Proposed commit message

In case of corrupted log file (which has good chances to happen in case of sudden unclean system shutdown), we set a flag which causes us to checkpoint immediately, but never do anything else besides that. This causes filebeat to just checkpoint on each log operation (therefore causing a high IO load on the server and also causing filebeat to fall behind).

This change resets the logInvalid flag after a successful checkpointing.


## Checklist




## Author's Checklist



## How to test this PR locally

I have not in fact tested the PR. I have only checked that it builds. I'm also not 100% sure of the correctness, but the change is really simple and I hope that a maintainer can quickly confirm whether the fix is correct. In any case the current code clearly will cause the issues I described, since `logInvalid` is only ever set to `true`, never to `false`. Therefore I think that the beat trivially cannot recover.

## Related issues

- Closes #35893

## Logs


See details in #35893 but the issue we fix otherwise manifests itself with a message of:
> WARN memlog/store.go:130 Incomplete or corrupted log file in /var/lib/filebeat/registry/filebeat. Continue with last known complete and consistent state. Reason: unexpected EOF

(or `invalid character '\x00'` instead of EOF)
